example/diamond.js

```diff
@@ -0,0 +1,331 @@
+
+			varying vec3 vWorldPosition;
+			varying vec3 vNormal;
+
+			uniform sampler2D envMap;
+			uniform float bounces;
+			uniform BVH bvh;
+			uniform float ior;
+			uniform vec3 color;
+			uniform bool correctMips;
+			uniform bool fastChroma;
+			uniform mat4 projectionMatrixInv;
+			uniform mat4 viewMatrixInv;
+			uniform mat4 modelMatrix;
+			uniform vec2 resolution;
+			uniform float aberrationStrength;
+
+			#include <cube_uv_reflection_fragment>
+
+			// performs an iterative bounce lookup modeling internal reflection and returns
+			// a final ray direction.
+			vec3 totalInternalReflection( vec3 incomingOrigin, vec3 incomingDirection, vec3 normal, float ior, mat4 modelMatrixInverse ) {
+
+				vec3 rayOrigin = incomingOrigin;
+				vec3 rayDirection = incomingDirection;
+
+				// refract the ray direction on the way into the diamond and adjust offset from
+				// the diamond surface for raytracing
+				rayDirection = refract( rayDirection, normal, 1.0 / ior );
+				rayOrigin = vWorldPosition + rayDirection * RAY_OFFSET;
+
+				// transform the ray into the local coordinates of the model
+				rayOrigin = ( modelMatrixInverse * vec4( rayOrigin, 1.0 ) ).xyz;
+				rayDirection = normalize( ( modelMatrixInverse * vec4( rayDirection, 0.0 ) ).xyz );
+
+				// perform multiple ray casts
+				for( float i = 0.0; i < bounces; i ++ ) {
+
+					// results
+					uvec4 faceIndices = uvec4( 0u );
+					vec3 faceNormal = vec3( 0.0, 0.0, 1.0 );
+					vec3 barycoord = vec3( 0.0 );
+					float side = 1.0;
+					float dist = 0.0;
+
+					// perform the raycast
+					// the diamond is a water tight model so we assume we always hit a surface
+					bvhIntersectFirstHit( bvh, rayOrigin, rayDirection, faceIndices, faceNormal, barycoord, side, dist );
+
+					// derive the new ray origin from the hit results
+					vec3 hitPos = rayOrigin + rayDirection * dist;
+
+					// if we don't internally reflect then end the ray tracing and sample
+					vec3 refractedDirection = refract( rayDirection, faceNormal, ior );
+					bool totalInternalReflection = length( refract( rayDirection, faceNormal, ior ) ) == 0.0;
+					if ( ! totalInternalReflection ) {
+
+						rayDirection = refractedDirection;
+						break;
+
+					}
+
+					// otherwise reflect off the surface internally for another hit
+					rayDirection = reflect( rayDirection, faceNormal );
+					rayOrigin = hitPos + rayDirection * RAY_OFFSET;
+
+				}
+
+				// return the final ray direction in world space
+				return normalize( ( modelMatrix * vec4( rayDirection, 0.0 ) ).xyz );
+			}
+
+			vec4 textureGradient( sampler2D envMap, vec3 rayDirection, vec3 directionCamPerfect ) {
+
+				// TODO: why are "correctMips" not being used? What's the point of this function?
+				vec2 uvv = equirectUv( rayDirection );
+				vec2 smoothUv = equirectUv( directionCamPerfect );
+				return texture( envMap, uvv, - 100.0 );//, dFdx(correctMips ? smoothUv : uvv), dFdy(correctMips ? smoothUv : uvv));
+
+			}
```

Comment:
@N8python Is there a reason you removed the use of `correctMips` here? And added -100 to the mip sampling?

Reply:
I was just trying to fix the mipmap aliasing bug r i p. You can change it back if you want.